treq/testing.py

```diff
@@ -205,3 +207,218 @@ def __init__(self, resource):
                 function = _reject_files(function)
 
             setattr(self, function_name, function)
+
+
+class StringStubbingResource(Resource):
+    """
+    A resource that takes a callable with 5 parameters
+    ``(method, url, params, headers, data) -> (code, headers, body)``
+    and uses the callable to return a real response as a result of a request.
+
+    The parameters for the callable are::
+
+        :param bytes method: An HTTP method
+        :param bytes url: The full URL of the request
+        :param dict params: A dictionary of query parameters mapping query keys
+            lists of values (sorted alphabetically)
+        :param dict headers: A dictionary of headers mapping header keys to
+            a list of header values (sorted alphabetically)
+        :param str data: The request body.
+        :return: a ``tuple`` of (code, headers, body) where the code is
+            the HTTP status code, the headers is a dictionary of bytes
+            (unlike the `headers` parameter, which is a dictionary of lists),
+            and body is a string that will be returned as the response body.
+
+    If there is a stubbing error, the return value is undefined (if an
+    exception is raised, :obj:`Resource` will just eat it and return 500
+    in its place).  The callable, or whomever creates the callable, should
+    have a way to handle error reporting.
+    """
+    isLeaf = True
+
+    def __init__(self, get_response_for):
+        """
+        :param get_response_for: A callable that takes 5 parameters:
```

Comment:
Rather than re-describing this parameter, I'd just say "```See ``StringStubbingResource``.```", since the parameter is already described in detail there.